From fa16211fff3ae55a858430b32e8f232157668c22 Mon Sep 17 00:00:00 2001
From: Xinyun Liu <xinyun.liu@intel.com>
Date: Thu, 24 May 2018 15:08:13 +0800
Subject: [PATCH 1042/1240] drm/i915/gvt: initialize the array to zero via
 compiler

'rxbuf' array elements are used uninitialized, it caused warning.
And make sure msg length longer than 4.

Signed-off-by: Xinyun Liu <xinyun.liu@intel.com>
Reviewed-by: Min He <min.he@intel.com>
Acked-by: Zhao Yakui <yakui.zhao@intel.com>
---
 drivers/gpu/drm/i915/gvt/edid.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/edid.c b/drivers/gpu/drm/i915/gvt/edid.c
index af06f63..a61546c 100644
--- a/drivers/gpu/drm/i915/gvt/edid.c
+++ b/drivers/gpu/drm/i915/gvt/edid.c
@@ -441,7 +441,7 @@ void intel_gvt_i2c_handle_aux_ch_write(struct intel_vgpu *vgpu,
 	u32 value = *(u32 *)p_data;
 	int aux_data_for_write = 0;
 	int reg = get_aux_ch_reg(offset);
-	uint8_t rxbuf[20];
+	uint8_t rxbuf[20] = {0};
 	size_t rxsize;
 
 	if (reg != AUX_CH_CTL) {
@@ -497,7 +497,8 @@ void intel_gvt_i2c_handle_aux_ch_write(struct intel_vgpu *vgpu,
 		 * operation to set block starting address
 		 */
 		if (addr == EDID_ADDR) {
-			i2c_edid->current_edid_read = rxbuf[4];
+			if (msg_length > 4)
+				i2c_edid->current_edid_read = rxbuf[4];
 		}
 	} else {
 		if (WARN_ON((op & 0x1) != GVT_AUX_I2C_READ))
-- 
2.7.4

